### PR TITLE
 added initial tests for agents in the-org

### PR DESCRIPTION
**Pull request by Y4NK33420**
# Relates to
 https://github.com/Sifchain/sa-eliza/issues/397

# Risks
Low - The changes are focused on test infrastructure and don't affect production code. The main risk is that some tests might fail in specific environments due to timing or process management issues.

# Background

## What does this PR do?
This PR implements an initial test suite for the ElizaOS agent system, establishing the foundation for future robust testing. The current implementation includes:
1. A basic test setup infrastructure with mock services
2. Initial social media manager agent tests
3. Basic structure for other agent-specific tests (Ruby, Project Manager, Investment Manager, Dev Rel, Community Manager)

Note: This is a draft PR with initial test implementations. The tests will be enhanced with more comprehensive scenarios, better error handling, and real-world use cases in future updates.

## What kind of change is this?
Improvements (misc. changes to existing features) - This PR adds initial test coverage for existing functionality without changing the core behavior.

# Documentation changes needed?
My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?
Reviewers should start by examining the test files in the `eliza/packages/the-org/tests/` directory:
1. `setup.ts` - Basic test infrastructure and mock services
2. `social-media-manager.test.ts` - Initial social media manager agent tests
3. Other agent-specific test files

## Detailed testing steps

1. **Test Setup Infrastructure**:
   - Run `bun test tests/setup.ts`
   - Verify basic mock scenario service functionality
   - Test message creation and user setup utilities
   - Verify runtime creation and cleanup

2. **Social Media Manager Tests**:
   - Run `bun test tests/social-media-manager.test.ts`
   - Test basic content creation requests
   - Test basic scheduling functionality
   - Test basic cross-platform campaign strategy
   - Test basic analytics interpretation
   - Test basic crisis communication
   - Test basic influencer collaboration

3. **Other Agent Tests**:
   - Run tests for each agent type:
     ```bash
     bun test tests/ruby.test.ts
     bun test tests/project-manager.test.ts
     bun test tests/investment-manager.test.ts
     bun test tests/dev-rel.test.ts
     bun test tests/community-manager.test.ts
     ```

## Database changes
None - These tests use mock services and don't affect the database.

## Deployment instructions
No special deployment instructions needed. The tests can be run locally using:
```bash
cd eliza/packages/the-org
bun test
```

## Future Improvements
The current test suite is a foundation that will be enhanced in future updates with:
1. More comprehensive test scenarios
2. Better error handling and edge cases
3. Real-world use cases and integration tests
4. Performance testing and benchmarks
5. More detailed assertions and validations
6. Better mock implementations for external services
7. Automated test data generation
8. Test coverage reporting and analysis

The current implementation provides a basic structure that can be built upon, focusing on establishing patterns and practices for future test development.

### CODE REVIEW

File: packages/the-org/package.json [hunk 1]
```diff
@@ -13,10 +13,19 @@
...
     "lint": "prettier --write ./src",
     "dev": "bun ../cli/dist/index.js dev",
-    "test": "bun ../cli/dist/index.js test",
+    "test": "vitest run",
     "clean": "rm -rf dist .turbo node_modules .turbo-tsconfig.json tsconfig.tsbuildinfo",
     "format": "prettier --write ./src",
-    "format:check": "prettier --check ./src"
+    "format:check": "prettier --check ./src",
+    "test:watch": "vitest",
+    "test:coverage": "vitest run --coverage",
```
Comment (monilpat) [inline packages/the-org/package.json:21]: can we add a test for test:org that runs all the tests in the org repo as well?

**Review by monilpat — COMMENTED** [review]
Overall the test structure makes sense to me would be curious to see the output of the tests to see what is passing and what is failing but imagine you are working through the failures now which is why it is in a draft state

**Comment by Y4NK33420** [comment]
Yeah the current structure includes use of mock implementations in few places which needs to be replaced with the actual class, but had some issues with them, will look into them and update the status.
